mongo/integration/cursor_test.go

```diff
@@ -42,4 +43,80 @@ func TestCursor(t *testing.T) {
 		ce := err.(mongo.CommandError)
 		assert.Equal(mt, int32(errorCursorNotFound), ce.Code, "expected error code %v, got %v", errorCursorNotFound, ce.Code)
 	})
+	mt.RunOpts("try next", noClientOpts, func(mt *mtest.T) {
+		mt.Run("existing non-empty batch", func(mt *mtest.T) {
+			// If there's already documents in the current batch, TryNext should return true without doing a getMore
+
+			initCollection(mt, mt.Coll)
+			cursor, err := mt.Coll.Find(mtest.Background, bson.D{})
+			assert.Nil(mt, err, "Find error: %v", err)
+			defer cursor.Close(mtest.Background)
+			tryNextExistingBatchTest(mt, cursor)
```

Comment:
Why don't you check `cursor.Next()` here the way you do for change streams?

Reply:
Because the `Find` will result in cursor with a non-empty first batch. It's different for change streams because calling `Watch` creates a change stream with an empty batch (unless a resume token pointing to an older time in the oplog is passed in) so the change stream test calls `Next` to get the first batch.